### PR TITLE
Raise errors from application endpoint with error codes

### PR DESCRIPTION
**Pull request by tomijarvi**
* Return full error detail from API Exception (detail + code)
* Raise 403 error from applicant validator with error code 1001 if the applicant has already applied to the project
* Fix the SSN validation on the application endpoint the validation raises an error if the SSN suffix is not valid. Added also error code 1000.
* Fix the SSN validation on the sales application endpoint that it uses the given profile instead of requester's profile. The salesperson can create applications on behalf of customers.

Error details and codes can be found in Confluence.

### CODE REVIEW

File: application_form/tests/test_application_api.py [hunk 1]
```diff
@@ -63,6 +63,38 @@ def test_application_post_writes_audit_log_if_not_authenticated(
...
+@pytest.mark.django_db
+def test_application_post_fails_if_incorrect_ssn_suffix(
+    api_client, elastic_single_project_with_apartments
+):
+    profile = ProfileFactory()
+    api_client.credentials(HTTP_AUTHORIZATION=f"Bearer {_create_token(profile)}")
+    data = create_application_data(profile)
+    data["ssn_suffix"] = "-000$"
+    response = api_client.post(
+        reverse("application_form:application-list"), data, format="json"
```
[the remark this answers is not in the document]
Comment (tomijarvi) [reply application_form/tests/test_application_api.py:75]: Will be fixed in another PR.

File: application_form/api/sales/serializers.py [hunk 2]
```diff
@@ -33,3 +35,20 @@ class Meta(ApplicationSerializer.Meta):
...
+        try:
+            validator(value)
+        except ValidationError as e:
+            message = f"""Invalid SSN suffix for the primary applicant was
+            received: {e.args[0]}"""
+            _logger.warning(message)
+            raise ValidationError(
+                detail=message,
+                code=1000,
+            )
```
[the remark this answers is not in the document]
Comment (tomijarvi) [reply application_form/api/sales/serializers.py:53]: This is a bit tricky as every validator is slightly different and so on the helper function might become a bit too complex. I would leave these now like this.

File: apartment_application_service/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            headers["WWW-Authenticate"] = exc.auth_header
+        if getattr(exc, "wait", None):
+            headers["Retry-After"] = "%d" % exc.wait
+
+        data = exc.get_full_details()
+
+        set_rollback()
+        return Response(data, status=exc.status_code, headers=headers)
+
+    return None
```
[the remark this answers is not in the document]
Comment (tomijarvi) [reply apartment_application_service/exceptions.py]: Changed

File: application_form/api/serializers.py [hunk 4]
```diff
@@ -38,7 +38,12 @@ def validate(self, attrs):
         try:
             validator(attrs.get("ssn_suffix", ""))
         except ValidationError as e:
-            _logger.warning("Invalid SSN suffix for applicant was received: %s", e)
+            message = f"Invalid SSN suffix for applicant was received: {e.args[0]}"
+            _logger.warning(message)
+            raise ValidationError(
+                detail={"ssn_suffix": message},
+                code=1000,
```
[the remark this answers is not in the document]
Comment (tomijarvi) [reply application_form/api/serializers.py]: Added constants

**Review by tuomas777 — APPROVED** [review]
LGTM 👍 🥇

**Pull request merged**